File: SortingAlgorithmsPython/insertionSort.py
def insertion_sort(arr):
    i = 1
    for step in range(1, len(arr)):
        i = step
        key = arr[i]
        for j in range(i - 1, -1, -1):
            if key < arr[j]:
                arr[i] = arr[j]
                arr[j] = key
                i -= 1
            else: break

File: SortingAlgorithmsCpp/SortingAlgorithms/InsertionSort.cpp
#include "InsertionSort.hpp"

void insertionSort(int arr[], int size) {
	int i = 1;
	for (int step = 1; step < size; step++) {
		i = step;
		int key = arr[i];
		for (int j = i - 1; j >= 0; j--) {
			if (key < arr[j]) {
				arr[i] = arr[j];
				arr[j] = key;
				i--;
			} else { break; }
		}
	}
}
